parquet-hadoop/src/main/java/org/apache/parquet/crypto/AesEncryptor.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.parquet.crypto;
+
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.GCMParameterSpec;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.apache.parquet.bytes.BytesUtils;
+import org.apache.parquet.format.BlockCipher;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.security.SecureRandom;
+import java.util.Arrays;
+
+public class AesEncryptor implements BlockCipher.Encryptor{
+
+  public enum Mode {
+    GCM, CTR
+  }
+  
+  // Module types
+  public static final byte Footer = 0;
+  public static final byte ColumnMetaData = 1;
+  public static final byte DataPage = 2;
+  public static final byte DictionaryPage = 3;
+  public static final byte DataPageHeader = 4;
+  public static final byte DictionaryPageHeader = 5;
+  public static final byte ColumnIndex = 6;
+  public static final byte OffsetIndex = 7;
+
+  public static final int NONCE_LENGTH = 12;
+  public static final int GCM_TAG_LENGTH = 16;
+  
+  static final int CTR_IV_LENGTH = 16;
+  static final int CHUNK_LENGTH = 4 * 1024;
+  static final int INT_LENGTH = 4;
+  static final int AAD_FILE_UNIQUE_LENGTH = 8;
+
+  private final SecretKey aesKey;
+  private final SecureRandom randomGenerator;
+  private final int tagLength;
+  private final Cipher aesCipher;
+  private final Mode aesMode;
+  private final byte[] ctrIV;
+  private final byte[] localNonce;
+
+  /**
+   * 
+   * @param mode GCM or CTR
+   * @param keyBytes encryption key
+   * @throws IllegalArgumentException
+   * @throws IOException
+   */
+  public AesEncryptor(Mode mode, byte[] keyBytes) throws IllegalArgumentException, IOException {
+    if (null == keyBytes) {
```

Comment:
Same as above. Should we check empty also?

Reply:
same comment as above